### PR TITLE
FOGL-39: logging async via postgres

### PR DESCRIPTION
**Pull request by amarendra-ScaleDB**
[FOGL-39](https://scaledb.atlassian.net/browse/FOGL-39)
This PR contains first pass of below tasks:

- Log to a table in Postgres. Each logged message has a 'code', a severity, and a message
- Define logging levels
- Tests added for postgres log
- "test" env now enabled

### CODE REVIEW

File: src/python/foglamp/foglamp_logger.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+logging.Logger.sensor1 = sensor1
+logging.Logger.sensor2 = sensor2
+
+foglamp_logger = logging.getLogger(__name__)
+foglamp_logger.setLevel(logging.DEBUG)
+
+foglamp_log_formatter = logging.Formatter('[%(asctime)s] %(levelname)s: %(name)s: %(message)s [%(filename)s/%(lineno)d %(module)s/%(funcName)s]')
+
+foglamp_filehandler = logging.FileHandler(FOGLAMP_DIR+'/fog.log')
```
Comment (praveen-garg) [inline src/python/foglamp/foglamp_logger.py]: should be configurable.. i.e. we may want: coap.log; rest.log, admin.log etc..

File: src/python/foglamp/postgres_handler.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    , sa.Column('created_by', sa.types.VARCHAR(50)))
+
+def log_msg(record):
+    """Construct log message from logging record components"""
+
+    msg = record.msg
+    msg = msg.strip()
+    msg = msg.replace('\'', '\'\'')
+
+    return msg
```
Comment (praveen-garg) [inline src/python/foglamp/postgres_handler.py]: there must be some way to dump as json

File: src/python/foglamp/foglamp_logger.py (hunk 1 above)
Comment (Terris-ScaleDB) [inline src/python/foglamp/foglamp_logger.py]: Files under the foglamp package should not contain the word 'foglamp' because it is redundant.

Comment (Terris-ScaleDB) [inline src/python/foglamp/foglamp_logger.py]: There will be potentially millions of sensors. We can't filter this way. The coap server would use some logger name that is hard-coded and would pass a dictionary to info/error/fatal/etc. which would be used by the logging module. The keys need to be defined.

File: src/python/foglamp/postgres_handler.py (hunk 2 above)
Comment (Terris-ScaleDB) [inline src/python/foglamp/postgres_handler.py]: All postgres access should be moved to a module in foglamp/model package

See https://scaledb.atlassian.net/browse/FOGL-128

**Comment by amarendra-ScaleDB** [comment]
Since logging requirements are still to be finalised and it seems that that more work will be needed in this task. The utility of this PR is more a kind of POC. Hence closing this PR.

**Pull request closed without merging**